runtime-sdk/modules/evm/src/precompile/erc20.rs

```diff
@@ -0,0 +1,586 @@
+use std::{marker::PhantomData, str::FromStr};
+
+use ethabi::{ParamType, Token};
+use evm::{
+    executor::stack::{PrecompileFailure, PrecompileHandle, PrecompileOutput},
+    ExitError, ExitSucceed,
+};
+use oasis_runtime_sdk::{
+    modules::accounts::{Error as AccountsError, API as AccountsAPI},
+    state::CurrentState,
+    storage,
+    types::{address, token},
+};
+use oasis_runtime_sdk_macros::{evm_contract_address, evm_method, sdk_derive, EvmEvent};
+use primitive_types::H160;
+
+use crate::precompile::{contract::EvmEvent as _, PrecompileResult};
+
+const MODULE_NAME: &str = "evm/erc20";
+const STATE_ALLOWANCES: &[u8] = &[0x01];
+
+/// Errors emitted by the ERC-20 implementation.
+#[derive(thiserror::Error, Debug, oasis_runtime_sdk_macros::Error)]
+pub enum Error {
+    #[error("forbidden by policy")]
+    #[sdk_error(code = 1)]
+    Forbidden,
+
+    #[error("invalid sender address {0}")]
+    #[sdk_error(code = 2)]
+    InvalidSender(H160),
```

Comment:
For errors that follow the EIP-6093 it would be great to follow the exact same naming, to make it clearer. Maybe also include a comment. Something like:

```suggestion
    /// EIP-6093 defined error.
    ERC20InvalidReceiver(H160),
```
Maybe the comment is not needed.


Btw how do these errors end up encoded? I guess just as strings "invalid sender address ..."?
Should we make it so that these predefined ERC20 errors end up encoded the same as in ERC-20 smart contracts? So I believe it should be: `keccak256("ERC20InvalidSender(address)")[:4]+<address>` for this one specifically?